kernel/src/main/java/org/kframework/compile/ConcretizeCells.java

```diff
@@ -69,4 +74,20 @@ public Sentence concretize(Module m, Sentence s) {
     s = sortCells.postprocess(s);
     return s;
   }
+
+  public static boolean hasCells(K item) {
+    if (IncompleteCellUtils.flattenCells(item).stream()
+        .anyMatch(k -> k.att().get(Production.class).att().contains(Att.CELL()))) {
+      return true;
+    }
+
+    if (item instanceof final KRewrite rew) {
```

Comment:
Pattern variables are implicitly `final` by design: https://docs.oracle.com/en/java/javase/14/docs/specs/patterns-instanceof-jls.html#jls-6.3.1.1

```suggestion
    if (item instanceof KRewrite rew) {
```